lxd/patches.go

```diff
@@ -854,40 +982,44 @@ func upgradeFromStorageTypeZfs(name string, d *Daemon, defaultPoolName string, d
 
 	if len(cRegular) > 0 {
 		containersSubvolumePath := getContainerMountPoint(poolName, "")
-		err := os.MkdirAll(containersSubvolumePath, 0711)
-		if err != nil {
-			return err
+		if !shared.PathExists(containersSubvolumePath) {
+			err := os.MkdirAll(containersSubvolumePath, 0711)
+			if err != nil {
+				return err
+			}
 		}
 	}
 
 	for _, ct := range cRegular {
-
 		// Insert storage volumes for containers into the database.
 		_, err := dbStoragePoolVolumeCreate(d.db, ct, storagePoolVolumeTypeContainer, poolID, volumeConfig)
 		if err != nil {
+			// In case we're on a LXD instance where a a partial
+			// upgrade took place, we should still proceed.
 			shared.LogWarnf("Could not insert a storage volume for container \"%s\".", ct)
```

Comment:
Should check that the error is because the entry already exists, if it is, then continue, if it's some other error, then fail.